cmd/syncthing/gui.go

```diff
@@ -1024,6 +1027,87 @@ func (s *apiService) getSystemLogTxt(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+type fileEntry struct {
+	name string
+	data []byte
+}
+
+func (s *apiService) getSupportBundle(w http.ResponseWriter, r *http.Request) {
+	var arrayFile []fileEntry
+
+	// Redacted configuration as a JSON
+	jsonConfig, _ := json.MarshalIndent(getRedactedConfig(s), "", "  ")
+	arrayFile = append(arrayFile, fileEntry{name: "config.json", data: jsonConfig})
+
+	// Log as a text
+	var buflog bytes.Buffer
+	for _, line := range s.systemLog.Since(time.Time{}) {
+		fmt.Fprintf(&buflog, "%s: %s\n", line.When.Format(time.RFC3339), line.Message)
```

Comment:
Again, perhaps we should do some annonimising by regexping IPs etc?

Reply:
Can we keep the IP as it is since it is seen at GUI, for the consistency ?

Reply:
I am not saying we have to anything at all, I am happy to send ips as they are, but I am saying that it might lack adoption if we don't annonimise stuff, as we have plenty of paranoid people using this.

@calmh?

Reply:
As @calmh suggestion I will implement redaction for device ID and IPs

Reply:
Lets do that in a separate pr tho.

Reply:
@calmh @AudriusButkevicius  What about the relevant lines at the existing Logs? Does this mean that we need to do same for the  existing Log implementation to cover the security usecase? I can see MyID at GUI -->  Actions --> Logs    If we change this at the current implementation, supportBundle does not have to go through logs and do mocking there since it will already be anonymous.

Reply:
People are concious of whats in the log as they see it when they copy it, they have no idea whats in the magic zip file they were asked to upload, unless they explicitly go in and check.

Reply:
I think that log viewer could be used for other purposes than just pasting logs for support. You could actually read the logs to debug something locally hence need to see ids and ips.

Reply:
Yeah, no redaction at the deeper levels. Only when creating support bundle. I'd probably do it regex based, IP numbers and device IDs are quite distinctive and can be detected that way.

I very much suggest a separate function to redact any given log line, and a unit test that runs a number of different log lines through it.